packages/components/managed-data-table/src/data-table-settings/column-manager/column-manager.spec.js

```diff
@@ -0,0 +1,232 @@
+      });
+
+      render(<ColumnManager {...props} />);
+
+      expect(screen.getByText(/search/i)).toBeInTheDocument();
+    });
+  });
+
+  describe('buttons', () => {
+    it('primary button', () => {
+      const primaryButtonLabel = 'Primary Test';
+      const props = createTestProps({
+        primaryButton: (
+          <PrimaryButton label={primaryButtonLabel} onClick={jest.fn()} />
+        ),
+      });
+      render(<ColumnManager {...props} />);
+      expect(screen.getByText(primaryButtonLabel)).toBeInTheDocument();
+    });
+
+    it('secondary button', () => {
+      const secondaryButtonLabel = 'Secondary Test';
+      const props = createTestProps({
+        secondaryButton: (
+          <SecondaryButton label={secondaryButtonLabel} onClick={jest.fn()} />
+        ),
+      });
+      render(<ColumnManager {...props} />);
+      expect(screen.getByText(secondaryButtonLabel)).toBeInTheDocument();
+    });
+  });
+
+  describe('handleDragEnd', () => {
+    describe('when destination and source is the hidden columns panel', () => {
+      it('should not call `onUpdateColumns`', () => {
+        const onUpdateColumns = jest.fn();
+        const setIsDragging = jest.fn();
+        const props = createTestProps({
+          selectedColumns: [
+            {
+              key: 'column1',
+              label: 'Column 1',
+            },
+          ],
+          availableColumns: [
+            {
+              key: 'column1',
+              label: 'Column 1',
+            },
+          ],
+          onUpdateColumns,
+        });
+
+        const dragResult = {
+          destination: { droppableId: HIDDEN_COLUMNS_PANEL },
+          source: { droppableId: HIDDEN_COLUMNS_PANEL },
+        };
+
+        handleDragEnd(
+          dragResult,
+          onUpdateColumns,
+          props.selectedColumns,
+          props.availableColumns,
+          setIsDragging
+        );
+
+        expect(onUpdateColumns).not.toHaveBeenCalled();
+      });
+    });
+
+    describe('when hidding columns', () => {
+      it('should call `onUpdateColumns` with updated selected columns', () => {
+        const onUpdateColumns = jest.fn();
+        const setIsDragging = jest.fn();
+        const props = createTestProps({
+          selectedColumns: [
+            {
+              key: 'column1',
+              label: 'Column 1',
+            },
+          ],
+          availableColumns: [
+            {
+              key: 'column1',
+              label: 'Column 1',
+            },
+          ],
+          onUpdateColumns,
+        });
+
+        const dragResult = {
+          destination: { droppableId: HIDDEN_COLUMNS_PANEL },
+          source: { droppableId: SELECTED_COLUMNS_PANEL, index: 0 },
+        };
+
+        handleDragEnd(
+          dragResult,
+          onUpdateColumns,
+          props.selectedColumns,
+          props.availableColumns,
+          setIsDragging
+        );
+
+        expect(onUpdateColumns).toHaveBeenCalledWith([]);
```

Comment:
Can we test with data that results in a non empty list? For example, having more than one item in the `selectedColumns` and make sure that the resulting list is without the dragged item.